includes/User.php

```diff
@@ -246,6 +247,11 @@ class User {
 	static $idCacheByName = array();
 
 	/**
+		* @var string the service auth token (currently helios); should NEVER be cached
+	 */
+	private $globalAuthToken = null;
```

Comment:
I bet that was happening in the `UserService`. If I remember correctly it was caching the whole user and not invalidating it. @owend fixed that about a week or so ago.